python/shotgun_globals/cached_schema.py

```diff
@@ -738,43 +738,63 @@ def get_status_color(cls, status_code, project_id=None):
         return status_color
 
     @classmethod
-    def field_is_editable(cls, sg_entity_type, field_name):
+    def field_is_editable(cls, sg_entity_type, field_name, project_id=None):
         """
         Returns a boolean identifying the editability of the entity's field.
 
         :param str sg_entity_type: the entity type
         :param str field_name: the field name to check editibility
+        :param project_id:  The id of the project.
+                            If None, the current context's project will be used.
```

Comment:
Looked at existing methods in the class and they say

```
        :param project_id:  The project Entity id. If None, the current
                            context's project will be used, or the "site"
                            cache location will be returned if the current
                            context does not have an associated project.
```

This seems good to me - maybe all we need to do is to make sure the docstring better reflect 
the actual logic in the case when a site context is being used?